docs/manual.md

```diff
@@ -978,6 +981,46 @@ Type annotations can be also applied to sets of records. For example:
 You can find more details on the simple type inference algorithm and the type
 annotations in [type annotations](types-and-annotations.md).
 
+### 7.2.3  Naturals
+
+If you look carefully at the [type annotations](types-and-annotations.md), you
+will find that there is no designated type for naturals. Indeed, one can just
+use the type `Int`, whenever a natural number is required. If we introduced a
+special type for naturals, that would cause a lot of confusion for the type
+checker. What would be the type of the literal `42`? That depends on, whether
+you extend `Naturals` or `Integers`. And if you extend `Naturals` and later
+somebody else extends your module and also `Integers`, should be the type
+of `42` be an integer?
+
+Apalache still allows you to extend `Naturals`. However, it will treat all
+number-like literals as integers. This is consistent with that the naturals are
+a subset of integers, and integers are a subset of reals.  Classically, one
+would not define subtraction for naturals. However, the module `Naturals`
```

Comment:
Is this right? Or would subtraction just be defined as a partial function? (Asking for my own edification here.)

Reply:
I also thought of Naturals!- as a partial function. But it is a total function. You can imagine that 4-10 returns an undefined value when applied to naturals, and this value happens to be -6 :-)

Reply:
Ah. Iiuc, you're saying that while `(-) : Nat x Nat -> Nat` is partial, we could also define it as `(-) : Nat x Nat -> Int` to make it total?

Reply:
Since there are no types in TLA+, `(-): Nat x Nat -> Nat` does not make a lot of sense in TLA+, at least for now. There is one `-` defined for Naturals, Integers, and Reals. But since we know that integers are closed under `-`, we pretend that there are no reals, when dealing with integers. The same trick applies to `+` with naturals.

Reply:
Ah, I see :) Thanks!